File: packages/bridge/src/utils/logger.ts
/**
 * Simple colored logger for the bridge CLI.
 */

const COLORS = {
  reset: "\x1b[0m",
  dim: "\x1b[2m",
  green: "\x1b[32m",
  yellow: "\x1b[33m",
  red: "\x1b[31m",
  cyan: "\x1b[36m",
  magenta: "\x1b[35m",
} as const;

export const log = {
  info: (msg: string, ...args: unknown[]) =>
    console.log(`${COLORS.cyan}[codepilot]${COLORS.reset} ${msg}`, ...args),

  success: (msg: string, ...args: unknown[]) =>
    console.log(`${COLORS.green}✓${COLORS.reset} ${msg}`, ...args),

  warn: (msg: string, ...args: unknown[]) =>
    console.log(`${COLORS.yellow}⚠${COLORS.reset} ${msg}`, ...args),

  error: (msg: string, ...args: unknown[]) =>
    console.error(`${COLORS.red}✗${COLORS.reset} ${msg}`, ...args),

  event: (sessionId: string, eventType: string, detail: string) =>
    console.log(
      `${COLORS.dim}[${sessionId.slice(0, 12)}]${COLORS.reset} ${COLORS.magenta}${eventType}${COLORS.reset} ${detail}`,
    ),

  connection: (msg: string) =>
    console.log(`${COLORS.green}📱${COLORS.reset} ${msg}`),
};

File: crates/codepilot-core/src/logger.rs
const RESET: &str = "\x1b[0m";
const DIM: &str = "\x1b[2m";
const GREEN: &str = "\x1b[32m";
const YELLOW: &str = "\x1b[33m";
const RED: &str = "\x1b[31m";
const CYAN: &str = "\x1b[36m";
const MAGENTA: &str = "\x1b[35m";

pub struct Logger;

impl Logger {
    pub fn info(&self, message: &str) {
        println!("{CYAN}[codepilot]{RESET} {message}");
    }

    pub fn success(&self, message: &str) {
        println!("{GREEN}\u{2713}{RESET} {message}");
    }

    pub fn warn(&self, message: &str) {
        println!("{YELLOW}!{RESET} {message}");
    }

    pub fn error(&self, message: &str) {
        eprintln!("{RED}x{RESET} {message}");
    }

    pub fn event(&self, session_id: &str, event_type: &str, detail: &str) {
        let trimmed = session_id.chars().take(12).collect::<String>();
        println!("{DIM}[{trimmed}]{RESET} {MAGENTA}{event_type}{RESET} {detail}");
    }

    pub fn connection(&self, message: &str) {
        println!("{GREEN}device{RESET} {message}");
    }
}

pub static LOG: Logger = Logger;
